sui-execution/latest/sui-adapter/src/temporary_store.rs

```diff
@@ -555,7 +555,16 @@ impl<'backing> TemporaryStore<'backing> {
                         None
                     }
                     Owner::ObjectOwner(_parent) => {
-                        unreachable!("Input objects must be address owned, shared, or immutable")
+                        unreachable!(
+                            "Input objects must be address owned, shared, consensus, or immutable"
+                        )
+                    }
+                    Owner::ConsensusV2 { authenticator, .. } => {
+                        assert!(
+                            sender == authenticator.as_single_owner(),
+                            "Sender must have permission to use input object"
+                        );
```

Comment:
i think it might be better to leave out any actual behavior right now in favor of `todo!()` - it would be much nicer to be able to see that code like this is going in at the same time as tests for it